Code/GraphMol/MolStandardize/Tautomer.h

```diff
@@ -40,29 +43,247 @@ inline int scoreTautomer(const ROMol &mol) {
 }
 }  // namespace TautomerScoringFunctions
 
+enum TautomerEnumeratorStatus {
+  Completed = 0,
+  MaxTautomersReached,
+  MaxTransformsReached,
+  Canceled
+};
+
+class Tautomer {
+  friend class TautomerEnumerator;
+
+ public:
+  Tautomer() : d_numModifiedAtoms(0), d_numModifiedBonds(0) {}
+  Tautomer(const ROMOL_SPTR &t, const ROMOL_SPTR &k, size_t a = 0, size_t b = 0)
+      : tautomer(t),
+        kekulized(k),
+        d_numModifiedAtoms(a),
+        d_numModifiedBonds(b) {}
+  ROMOL_SPTR tautomer;
+  ROMOL_SPTR kekulized;
+
+ private:
+  size_t d_numModifiedAtoms;
+  size_t d_numModifiedBonds;
+};
+
+typedef std::map<std::string, Tautomer> SmilesTautomerMap;
+typedef std::pair<std::string, Tautomer> SmilesTautomerPair;
+
+//! Contains results of tautomer enumeration
+class RDKIT_MOLSTANDARDIZE_EXPORT TautomerEnumeratorResult {
+  friend class TautomerEnumerator;
+
+ public:
+  class const_iterator {
+   public:
+    typedef ROMOL_SPTR value_type;
+    typedef std::ptrdiff_t difference_type;
```

Comment:
value_type, difference_type aren't used are they?

Reply:
They aren't, but they are part of the `std::iterator_traits` that should be defined for every iterator:
http://www.cplusplus.com/reference/iterator/iterator_traits/
This ensures, for example, that `std::distance` can be used on this iterator; I have just added tests for that.